src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php

```diff
@@ -113,4 +113,36 @@ private function executeSearchQuery($expectedResults = 1)
 
         return $results;
     }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRename()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin');
+
+        $result = $this->executeSearchQuery(1);
+        $entry = $result[0];
+        $this->assertEquals($entry->getAttribute('cn')[0], 'Kevin');
+    }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRenameWithoutRemovingOldRdn()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin', false);
+
+        $this->executeSearchQuery(2);
```

Comment:
Maybe instead of `$this->executeSearchQuery(2)`, try to copy part of the logic, and use something to print what you have in the test.

My guess is that there is something wrong with the parent DN or the RDN, which may not be set to the previous value.

I would recommend you try to run the tests locally (by using the `csarrazi/symfony-ldap` docker image, for example), and check what you have in the LDAP directory using `var_dump()`, for example.

FYI, the container can be run as follows:

```
docker run -d --name ldap -p 389:389 -v "/path/to/symfony/source/code/src/Symfony/Component/Ldap/Tests/Fixtures/data:/init-ldap.d" -e "LDAP_SUFFIX=dc=symfony,dc=com" -e "LDAP_ROOTDN=cn=admin,dc=symfony,dc=com" -e "LDAP_ROOTPW=symfony" csarrazi/symfony-ldap
```

Then you can run the tests as follows (from `/path/to/symfony/source/code`):

```
LDAP_HOST=localhost LDAP_PORT=389 ./phpunit src/Symfony/Component/Ldap
```

Cheers!